File: misanuncios-ui/src/app/infrastructure/dto/userHasPermission.dto.ts
import { PermissionDto } from "./permission.dto";
import { UserDto } from "./user.dto";
import { UserHasPermissionKeyDto } from "./userHasPermissionKey.dto";

export interface UserHasPermissionDto {

  id: UserHasPermissionKeyDto;
  user: UserDto; // Assuming User model exists
  permission: PermissionDto; // Assuming Permission model exists


}

File: misanuncios/src/main/java/com/yaagoub/misanuncios/infrastructure/rest/spring/dto/UserHasPermissionDto.java
package com.yaagoub.misanuncios.infrastructure.rest.spring.dto;

import com.fasterxml.jackson.annotation.JsonIdentityInfo;
import com.fasterxml.jackson.annotation.ObjectIdGenerators;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.ToString;

@Data
@EqualsAndHashCode(exclude = { "id","user","permission"})
public class UserHasPermissionDto {
    private UserHasPermissionKeyDto id;

    private UserDto user;

    private PermissionDto permission;
}
